lib/Models/ArcGisFeatureServerCatalogItem.ts

```diff
@@ -0,0 +1,621 @@
+    geoJsonItem.setTrait(
+      CommonStrata.definition,
+      "clampToGround",
+      item.clampToGround
+    );
+
+    return loadGeoJson(item)
+      .then(geoJsonData => {
+        geoJsonItem.setTrait(
+          CommonStrata.definition,
+          "geoJsonData",
+          geoJsonData
+        );
+        return geoJsonItem.loadMetadata();
+      })
+      .then(() => {
+        return loadMetadata(item);
+      })
+      .then(featureServer => {
+        const stratum = new FeatureServerStratum(
+          item,
+          geoJsonItem,
+          featureServer
+        );
+        return stratum;
+      });
+  }
+
+  @computed get maximumScale() {
+    return this._featureServer.maxScale;
+  }
+
+  @computed get name() {
+    if (this._featureServer.name && this._featureServer.name.length > 0) {
+      return this._featureServer.name;
+    }
+  }
+
+  @computed get dataCustodian() {
+    if (
+      this._featureServer.documentInfo &&
+      this._featureServer.documentInfo.author &&
+      this._featureServer.documentInfo.author.length > 0
+    ) {
+      return this._featureServer.documentInfo.author;
+    }
+  }
+
+  @computed get rectangle() {
+    const extent = this._featureServer.extent;
+
+    if (
+      isDefined(extent) &&
+      extent.spatialReference &&
+      extent.spatialReference.wkid
+    ) {
+      const wkid = "EPSG:" + extent.spatialReference.wkid;
+      if (!isDefined((proj4definitions as any)[wkid])) {
+        return undefined;
+      }
+
+      const source = new proj4.Proj((proj4definitions as any)[wkid]);
+      const dest = new proj4.Proj("EPSG:4326");
+
+      let p = proj4(source, dest, [extent.xmin, extent.ymin]);
+
+      const west = p[0];
+      const south = p[1];
+
+      p = proj4(source, dest, [extent.xmax, extent.ymax]);
+
+      const east = p[0];
+      const north = p[1];
+
+      const rectangle = { west: west, south: south, east: east, north: north };
+      return createStratumInstance(RectangleTraits, rectangle);
+    }
+
+    return undefined;
+  }
+
+  @computed get info() {
+    function newInfo(name: string, content?: string) {
+      const traits = createStratumInstance(InfoSectionTraits);
+      runInAction(() => {
+        traits.name = name;
+        traits.content = content;
+      });
+      return traits;
+    }
+
+    return [
+      newInfo(
+        i18next.t("models.arcGisMapServerCatalogItem.dataDescription"),
+        this._featureServer.description
+      ),
+      newInfo(
+        i18next.t("models.arcGisMapServerCatalogItem.copyrightText"),
+        this._featureServer.copyrightText
+      )
+    ];
+  }
+
+  @computed get legends(): StratumFromTraits<LegendTraits>[] | undefined {
+    if (
+      !this._item.useStyleInformationFromService ||
+      !this._featureServer.drawingInfo
+    ) {
+      return undefined;
+    }
+    const renderer = this._featureServer.drawingInfo.renderer;
+    const rendererType = renderer.type;
+
+    if (rendererType === "classBreaks") {
```

Comment:
Probably need to work out how to render legends for other renderer types like `uniqueValue`. This service offers a good example
https://services.arcgis.com/P3ePLMYs2RVChkJx/ArcGIS/rest/services/World_Regions/FeatureServer/0 , and here's an associated [Esri demo map](https://www.arcgis.com/home/webmap/viewer.html?url=https://services.arcgis.com/P3ePLMYs2RVChkJx/ArcGIS/rest/services/World_Regions/FeatureServer/0&source=sd)